btf/testdata/tags.c

```diff
@@ -0,0 +1,33 @@
+#include "../../testdata/common.h"
+
+#define tagA __attribute__((btf_decl_tag("a")))
+#define tagB __attribute__((btf_decl_tag("b")))
+#define tagC __attribute__((btf_decl_tag("c")))
+#define tagD __attribute__((btf_decl_tag("d")))
+#define tagE __attribute__((btf_decl_tag("e")))
+
+struct s {
+	char tagA foo;
+	char tagB bar;
+} tagC;
+
+union u {
+	char tagA foo;
+	char tagB bar;
+} tagC;
+
+typedef tagB char td;
+
+struct s tagD s1;
+union u tagE u1;
+td tagA t1;
+
+int tagA tagB fwdDecl(char tagC x, char tagD y);
+
+int tagE normalDecl(char tagB x, char tagC y) {
+	return fwdDecl(x, y);
+}
+
```

Comment:
Would it make sense to declare two functions with an identical prototype here that could one day be deduplicated by llvm? Might be useful as a canary.